src/components/Layout.js

```diff
@@ -50,6 +58,35 @@ export default function Layout({
   )
 }
 
+let accountSettingsTimer = null
+
+/**
+ * make sure we have all the account settings for current user
+ * @param {Object} authentication
+ * @param {boolean} showAccountSetup
+ * @param {string} languageId
+ * @param {string} owner
+ * @param {function} setShowAccountSetup
+ */
+function verifyAccountSettings(authentication, showAccountSetup, languageId, owner, setShowAccountSetup) {
```

Comment:
This one should also be converted to a custom hook

Reply:
sounds good